src/main/scala/com/typesafe/sbt/packager/archetypes/jlink/JlinkPlugin.scala

```diff
@@ -102,4 +150,43 @@ object JlinkPlugin extends AutoPlugin {
     private def list(arg: String, values: Seq[String]): Seq[String] =
       if (values.nonEmpty) Seq(arg, values.mkString(",")) else Nil
   }
+
+  // Jdeps output row
+  private final case class PackageDependency(dependent: String, dependee: String, source: PackageDependency.Source)
+
+  private final object PackageDependency {
+    sealed trait Source
+
+    object Source {
+      def parse(s: String): Source = s match {
+        case "not found" => NotFound
+        // We have no foolproof way to separate jars from modules here, so
+        // we have to do something flaky.
+        case name
+            if name.toLowerCase.endsWith(".jar") ||
+              !name.contains('.') ||
+              name.contains(' ') =>
+          JarOrDir(name)
+        case name => Module(name)
+      }
+    }
+
+    case object NotFound extends Source
+    final case class Module(name: String) extends Source
+    final case class JarOrDir(name: String) extends Source
+
+    private val pattern = """^\s+([^\s]+)\s+->\s+([^\s]+)\s+([^\s].*?)\s*$""".r
```

Comment:
Can you add examples for some common patterns that occur?

Reply:
Done.